.packit.yaml

```diff
@@ -36,3 +40,70 @@ jobs:
       # do not get the version from a tag (git describe) but from the spec file
       get-current-version:
         - grep -oP '^Version:\s+\K\S+' packaging/command-line-assistant.spec
+
+  # Test jobs
+  - &tests-e2e-vm
+    job: tests
+    identifier: vm-e2e-tests
+    # Run tests on-demand
+    manual_trigger: true
+    trigger: pull_request
+    fmf_url: "https://gitlab.cee.redhat.com/rhel-lightspeed/enhanced-shell/cla-tests"
+    fmf_ref: main
+    # TODO: Replace with tmt_plan: "plans/e2e" or similar
+    tmt_plan: "plans"
+    use_internal_tf: True
+    targets:
+      epel-9-x86_64:
+        distros: [RHEL-9-Nightly]
+      epel-10-x86_64:
+        distros: [RHEL-10-Nightly]
+      epel-9-aarch64:
+        distros: [RHEL-9-Nightly]
+      epel-10-aarch64:
+        distros: [RHEL-10-Nightly]
+    tf_extra_params:
+      environments:
+        - settings:
+            provisioning:
+              tags:
+                BusinessUnit: sst_rspeed_qe
+    labels:
+      - e2e
+      - vm
+
+  - &tests-sanity-container
+    job: tests
+    <<: *tests-e2e-vm
+    identifier: container-sanity-tests
+    manual_trigger: false
+    # TODO: Replace with tmt_plan: "plans/sanity"
+    tmt_plan: "plans"
+    targets:
+      # There is currently only RHEL-9 container available at the moment
+      epel-9-x86_64:
+        distros: [RHEL-9-Nightly]
+    tf_extra_params:
+      environments:
+        - os: null
+          variables:
+            IMAGE: registry.access.redhat.com/ubi9:latest
+          settings:
+            provisioning:
+              tags:
+                BusinessUnit: sst_rspeed_qe
+    labels:
+      - sanity
+      - sanity-container
+      - container
```

Comment:
```suggestion
```

I'd remove this section as we're testing nightlies in the next one.

Reply:
This section is for provisioning the container and running the basic sanity tests in it. The RHEL-9 Nightly is there only because there needs to be something (packit doesn't fully support requesting the container, so this is a bit of a workaround/hack to actually run a test in a container). I want to keep the container for the fast test execution.

Reply:
The current version of `ubi9:latest` is RHEL 9.4:
`podman run --rm -it registry.access.redhat.com/ubi9:latest cat /etc/os-release`

So I don't think we would be testing in an environment that resembles the targeted release, and this might lead to the tests incorrectly failing. This would flag the PR tests as failed and incorrectly warn the developer about a failure, making the PR not mergeable.

Also, we'd need to wait for the Nightly tests to be completed before the tests are marked as passed, so the container tests won't necessarily speed up the process. 

If we want to keep it, would there be a way that a failure in container tests does not affect the global pass/fail result of the checks? Or maybe using more up-to-date container images?

Reply:
For the container thing. I had this PR workflow in mind:

1. dev creates a PR
2. automatically for each commit/push to the PR, basic unity tests will run together with the really basic sanity tests. By that I mean really fast first tests like `c --help`, `c --config "something"`, "running c as a root", and tests like these. For that, I do not think that having correct target release is that important (maybe with the release of rhel-9.6 we will get the 9.6 container image). 
3. once the dev thinks that the PR is ready for the whole test suite, he can manually trigger the whole test suite, which would the remaining tests and ALSO run the basic sanity test suite again as in the container (this time on correct distro and arch), so we will not skip the proper verification.

The reason is that the first initial container tests can show early on, if some basic functionality is broken. Yes, in the end the dev will have to run the whole test suite (or the one using the VM) but he doesn't have to do it early. In c2r we had to reduce the test suite to have some basic sanity tests that are much faster than the whole test suite as it was getting too slow to get some results. 

If for some reason, the sanity tests will fail on container and not in the regular VM we can probably remove it or just somehow re-work this.

Reply:
I'm OK for having super fast sanity tests for commits, but the fact that step 3 needs to be manually run and not automatically run before PR reviews is something that could make some errors slip inadvertently to the `main` branch, only to be found on downstream tests.

So if it's possible to make step 3 automatic (PRs sent for review would have tests automatically triggered) then I'd be OK with this approach as it'd fulfill the shift-left goal of upstream tests. We'd also need to make sure the cost of running tests for every commit are acceptable.

Reply:
> So if it's possible to make step 3 automatic (PRs sent for review would have tests automatically triggered) then I'd be OK with this approach as it'd fulfill the shift-left goal of upstream tests. We'd also need to make sure the cost of running tests for every commit are acceptable.

I'll try to investigate on having this as either a mandatory check that needs to pass, or to have some automation that will trigger the tests.

Reply:
Ok, it seems that the packit is now showing the triggered jobs (https://github.com/packit/packit-service/issues/2147). So we (and the devs) have to always check if the manual tests were triggered as well. Running them automatically for each commit could cause that we run out of quota very fast

Reply:
however, we can set the packit as a mandatory check before merging